File: ASCOM.Utilities/ASCOM.Utilities.Support/JsonResponseClasses/AlpacaDeviceDescription.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ASCOM.Utilities.Support
{
    public class AlpacaDeviceDescription
    {
        public AlpacaDeviceDescription() { }

        public AlpacaDeviceDescription(string serverName, string manufacturer, string manufacturerVersion, string location, string alpacaUniqueId)
        {
            ServerName = serverName;
            Manufacturer = manufacturer;
            ManufacturerVersion = manufacturerVersion;
            Location = location;
            AlpacaUniqueId = alpacaUniqueId;
        }

        public string ServerName { get; set; } = "";
        public string Manufacturer { get; set; } = "";
        public string ManufacturerVersion { get; set; } = "";
        public string Location { get; set; } = "";
        public string AlpacaUniqueId { get; set; } = "";
    }
}

File: ASCOM.Utilities/ASCOM.Utilities/AlpacaSupport/JsonResponseClasses/AlpacaDeviceDescription.vb
Namespace ASCOM.Utilities.Support
    Public Class AlpacaDeviceDescription
        Public Sub New()
        End Sub

        Public Sub New(ByVal serverName As String, ByVal manufacturer As String, ByVal manufacturerVersion As String, ByVal location As String, ByVal alpacaUniqueId As String)
            Me.ServerName = serverName
            Me.Manufacturer = manufacturer
            Me.ManufacturerVersion = manufacturerVersion
            Me.Location = location
            Me.AlpacaUniqueId = alpacaUniqueId
        End Sub

        Public Property ServerName As String = ""
        Public Property Manufacturer As String = ""
        Public Property ManufacturerVersion As String = ""
        Public Property Location As String = ""
        Public Property AlpacaUniqueId As String = ""
    End Class
End Namespace
